### ISSUE TITLE
Home directory is cluttered with mpv shader cache files

### ISSUE THREAD

**Issue by oficsu** [mpv-player/mpv#12418]
### Important Information

- mpv 0.36.0 
- manjaro linux
- [extra/mpv](https://software.manjaro.org/package/mpv) 
- kde 5.27.7

### Reproduction steps

```bash
mkdir ./{config,cache}
curl -O https://raw.githubusercontent.com/mpv-player/mpv.io/master/source/images/mpv-logo-128.png
mpv --log-file=./output.txt --cscale=spline36 --cache-dir=./cache --config-dir=./config mpv-logo-128.png
```

### Expected behavior

The `./config` directory isn't cluttered with a bunch of obscure hex-named mpv shader cache files

### Actual behavior

It is:
```
$ tree
.
├── cache
├── config
│   ├── 7D6817E3C130F0362528531171DAA52F702D0D15A05362392563AEED17651368
│   ├── 869A8CECAD6235F2E3D650580694A6C756F14D0AD0CFD6572533C729C4366736
│   └── B99F3F15D25F561D070BB394B8A85526DBDB9E4C8886A8CEC6D06ECB6DCF0714
└── mpv-logo-128.png
```

### Log file

[output.txt](https://github.com/mpv-player/mpv/files/12644513/output.txt)

### Sample files

I used the [mpv logo](https://raw.githubusercontent.com/mpv-player/mpv.io/master/source/images/mpv-logo-128.png), but the same with other videos/images

### Notes

I think these files _can_ go into `./cache` directory, but I'm sure they must never be stored in the `./config` directory

**Comment by Dudemanguy** [comment]
It's because you're passing `--config-dir=./config` which overrides the location of any files mpv makes and puts them in config.

**Comment by oficsu** [comment]
Firstly, it's not just a bug, it's a UX issue. It's completely broken UX.

Secondly,

> It's because you're passing --config-dir=./config which overrides the location of any files mpv makes and puts them in config

I don't see the reason why they aren't aware of the explicitly specified `--cache-dir`:
```
--cache-dir=<path>
  Directory where to create temporary files. Cache is stored in the
  system's cache directory (usually ˜/.cache/mpv) if this is unset.

  Currently, this is used for --cache-on-disk only.
```

And even more, when you define `$XDG_CACHE_HOME`:
```bash
XDG_CACHE_HOME=./cache 
  mpv --log-file=output.txt \
      --cscale=spline36 \
      --cache-dir="./cache" \
      --config-dir="./config" \
      mpv-logo-128.png
```
there are a lot of cache files in the cache directory for `fontconfig` and `mesa_shader_cache`. But "mpv shader **cache** files" are saved in the config directory. Regardless of mpv implementation details, since these files are cache files, they should follow [XDG specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html):
> `$XDG_CACHE_HOME` defines the base directory relative to which user-specific non-essential data files should be stored

**Comment by Dudemanguy** [comment]
`--cache-dir` is a horribly named option and has nothing to do with gpu shader cache. But regardless, using the `--config-dir` option overrides everything.

**Comment by Dudemanguy** [comment]
Fixed by #12541.
